lxd/main_forkproxy.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/shared"
+)
+
+/*
+#define _GNU_SOURCE
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+extern bool advance_arg(char *buf, char *cur, ssize_t size, bool required);
+extern int dosetns(int pid, char *nstype);
+
+void forkproxy(char *buf, char *cur, ssize_t size) {
+	int cmdline, listen_pid, connect_pid, fdnum, forked, childPid, ret;
+	char fdpath[80];
```

Comment:
I know, this is complaining about details and this is not your addition but we should always justify why a buffer was chosen at a fixed size for posterity. It seems that this is supposed to be a `/proc/<pid>/fd/<idx>` path. So an appropriate macro would be:

```
/*
 * strlen("/proc") = 5
 * +
 * 64 bit int to chars = 2^64 - 1 = 21 chars
 * +
 * strlen("/fd") = 3
 * +
 * 64 bit int to chars = 2^64 - 1 = 21 chars
 * +
 * \0 = 1
 */

 #define __PROC_FD_LEN 51
char fdpath[80];
```

Reply:
Actually, this is /proc/self/fd/NUM, so should just be: 14 + 21 +1

Reply:
fixed